### PR TITLE
refactor: frontend

### PR DESCRIPTION
**Pull request by gyroflaw**
Fix Refactor Frontend #112
### 😤 Problem / Goals
-

### 🤓 Solution
- Revert  and use react-app-rewired to make simple
- Replace emotion with styled-component

### 🗒️ Additional Notes
-

### CODE REVIEW

**Comment by hoffmannjan** [comment]
Just did a overall look but I'm not sure do we want to revert `eject` now. @DorianKwan any opinions?

**Comment by DorianKwan** [comment]
I'm confused why we are reverting back to CRA setup.  We are losing control of a large set of configuration that doesn't lend well to the maintenance and development of a production application.

I don't believe reverting the the eject is the way to go.

For context, we ejected with the long term goal of stripping away all CRA setup in favour of a more custom setup.  We didn't have the time to remove it completely yet.  And that's what led us to today.  Where the eject doesn't seem like it's worth it at the moment in favour of rewired.  But the decision to not use rewired has previously been made.

Can you expand on your idea of using rewired instead?

File: frontend/.eslintrc.js [hunk 1]
```diff
@@ -3,116 +3,5 @@ module.exports = {
...
-    'import/resolver': {
-      node: {
-        moduleDirectory: ['node_modules', 'src/'],
-      },
-    },
-    react: {
-      version: 'detect',
-    },
-  },
+  extends: ['react-app', 'react-app/jest', 'prettier'],
```
Comment (DorianKwan) [inline frontend/.eslintrc.js:6]: There is also a plan in the future to create our own package to keep eslint configurations consistent project to project.  Let's not change this completely.  If you want to add or change a few rules let's chat about that.

File: frontend/src/components/base/Button/Button.test.tsx [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
 import userEvent from '@testing-library/user-event';
 import { render } from '../../../test-utils';
 import { Button } from './Button';
 
 describe('Button', () => {
   it('should render a button', () => {
-    const { getByRole } = render(<Button type="button">button</Button>);
-    const button = getByRole('button', { name: 'button' });
+    render(<Button type="button">button</Button>);
+    const button = screen.getByRole('button', { name: 'button' });
```
Comment (DorianKwan) [inline frontend/src/components/base/Button/Button.test.tsx:10]: Is there any concerns here with not using the method off of the custom render?

**Review by DorianKwan — CHANGES_REQUESTED** [review]

**Comment by hoffmannjan** [comment]
Hi @gyroflaw, for now please let's stick to ejected project. 
We can decide about this later on in January, there are too many question marks right now about project future so I don't want anybody to spend time on this. You can keep it as a branch that we can use in future.

**Comment by hoffmannjan** [comment]
Oh and there are now a few conflicts because @valfaro123 added multilang support, please look at them, then you should be unlocked. Cheers :)

**Comment by hoffmannjan** [comment]
@gyroflaw please revert React app eject and resolve the conflicts so we can merge it.

**Review by hoffmannjan — CHANGES_REQUESTED** [review]
@gyroflaw please revert React app eject and resolve the conflicts so we can merge it.

**Pull request closed without merging**